autoload/neomake.vim

```diff
@@ -504,41 +510,45 @@ function! neomake#MakeHandler(job_id, data, event_type) abort
     let jobinfo = s:jobs[a:job_id]
     let maker = jobinfo.maker
     if index(['stdout', 'stderr'], a:event_type) >= 0
-        let lines = a:data
-        if has_key(maker, 'mapexpr')
-            let lines = map(copy(lines), maker.mapexpr)
-        endif
-
-        for line in lines
-            call neomake#utils#DebugMessage(
-                \ get(maker, 'name', 'makeprg').' '.a:event_type.': '.line)
-        endfor
+        call neomake#utils#DebugMessage(
+            \ get(maker, 'name', 'makeprg').' '.a:event_type.': ["'.join(a:data, '", "').'"]')
         call neomake#utils#DebugMessage(
             \ get(maker, 'name', 'makeprg').' '.a:event_type.' done.')
 
         " Register job output. Buffer registering of output for long running
         " jobs.
         let last_event_type = get(jobinfo, 'event_type', a:event_type)
         let jobinfo.event_type = a:event_type
+
+        " a:data is a List of 'lines' read. Each element *after* the first
+        " element represents a newline
         if has_key(jobinfo, 'lines')
             " As per https://github.com/neovim/neovim/issues/3555
             let jobinfo.lines = jobinfo.lines[:-2]
-                        \ + [jobinfo.lines[-1] . get(lines, 0, '')]
-                        \ + lines[1:]
+                        \ + [jobinfo.lines[-1] . get(a:data, 0, '')]
+                        \ + a:data[1:]
         else
-            let jobinfo.lines = lines
+            let jobinfo.lines = a:data
+        endif
+
+        if len(a:data) < 2
+            return
```

Comment:
I'm embarrassed to admit I don't remember. I think it is just an optimization for the following change. If there are fewer than 2 lines in `a:data` then calling `sRegisterJobOutput(jobinfo, maker, jobinfo.lines[:-2])` won't do anything since `jobinfo.lines[:-2]` will be empty. Though, it looks like there's a bug since it should be checking `len(jobinfo.lines) < 2` instead :/

I think it can just get removed. Thoughts?